src/cbmc/cbmc_solvers.cpp

```diff
@@ -169,8 +171,10 @@ std::unique_ptr<cbmc_solverst::solvert> cbmc_solverst::get_smt2(
   {
     if(solver==smt2_dect::solvert::GENERIC)
     {
-      error() << "please use --outfile" << eom;
-      throw 0;
+      throw invalid_user_input_exceptiont(
+        "Failed to provide a filename",
```

Comment:
Who failed?

Reply:
The user, but this is implicit in the error message showed by the `what()` function of this exception, which will look like this:

```
Invalid User Input
Option: --filename
Reason: Failed to provide a filename
```

Reply:
But I guess there's also an issue with the error string, it should start with a lowercase letter.

Reply:
No, users never fail, by definition. You can ask them to do something different ("Please provide a filename"), or you can firmly order them to do so ("Filename required"/"Required filename not provided").

Reply:
I think I will go with the last string, looks more reasonable in this context in my opinion `"Required filename not provided"`